### PR TITLE
Signals: Quandl example script

### PR DESCRIPTION
**Pull request by parmarsuraj99**
Quandl example script for Signals and README in a new folder

### CODE REVIEW

File: signals/quandl/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+This script uses Quandl's [End of Day US Stock Prices](https://www.quandl.com/data/EOD-End-of-Day-US-Stock-Prices/) data. (It's a premium data source)
+
+Two ways to download the data
+
+1. Load every ticker and its daily features sequentially. (slower)
+2. Load entire dataset once and load data from it. (much faster)
+
+This script provides functions for both ways of loading the data in same format.
```
Comment (LiamHz) [inline signals/quandl/README.md:8]:
```suggestion
This script uses Quandl's [end of day US stock prices](https://www.quandl.com/data/EOD-End-of-Day-US-Stock-Prices/) data. 
```

File: signals/quandl/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+*Need to install quandl's python package and numerapi*
+
+```
+pip install quandl
+pip install numerapi
+```
+
+**`API_KEY = "<Quandl API KEY>"` This need to be set before running the script.**
```
Comment (LiamHz) [inline signals/quandl/README.md:17]:
```suggestion
**`API_KEY = "<Quandl API KEY>"` This needs to be set before running the script.**

To get an API key, create a paid account on Quandl.
```

File: signals/quandl/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+This script uses Quandl's [End of Day US Stock Prices](https://www.quandl.com/data/EOD-End-of-Day-US-Stock-Prices/) data. (It's a premium data source)
+
+Two ways to download the data
+
+1. Load every ticker and its daily features sequentially. (slower)
+2. Load entire dataset once and load data from it. (much faster)
+
+This script provides functions for both ways of loading the data in same format.
+
+*Need to install quandl's python package and numerapi*
```
Comment (LiamHz) [inline signals/quandl/README.md:10]:
```suggestion
*Before running, install quandl's python package and numerapi*
```

File: signals/quandl/example_model_quandl.py [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    full_data.sort_index(ascending=True, inplace=True)
+    gc.collect()
+
+    return full_data
+
+
+def main():
+
+    # downloads the whole dataset as zip and read data (takes around 1.5min)
+    # Similar to loading data using yfinance in main example_model.py
```
Comment (LiamHz) [inline signals/quandl/example_model_quandl.py:129]:
```suggestion
def main():
    # downloads the whole dataset as zip and read data (takes around 1.5min)
```

File: signals/quandl/example_model_quandl.py [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+                f"{indicator}_quintile"
+            ].transform(lambda group: group.shift(day))
+
+        gc.collect()
+
+    full_data.dropna(axis=0, inplace=True)
+
+    del ticker_groups
+    gc.collect()
+
```
Comment (LiamHz) [inline signals/quandl/example_model_quandl.py:196]:
```suggestion

```

**Review by LiamHz — CHANGES_REQUESTED** [review]
Made a few small changes. Once they're made this PR looks good to merge

File: signals/quandl/example_model_quandl.py [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+eod_tickers = pd.read_csv(
+    "https://s3.amazonaws.com/quandl-production-static/end_of_day_us_stocks/ticker_list.csv"
+)
+print(f"Number of eligible tickers : {len(eligible_tickers)}")
+
+common_tickers = np.intersect1d(
+    yfinance_tickers.values.astype(str), eod_tickers["Ticker"].values.astype(str)
+)
+print(f"Number of tickers common between EOD and Bloomberg: {len(common_tickers)}")
```
[the remark this answers is not in the document]
Comment (parmarsuraj99) [reply signals/quandl/example_model_quandl.py:44]: Yes, Will pass the `common_tickers` and `ticker_map` as arguments to the `download_full_and_load` function

File: signals/quandl/example_model_quandl.py (hunk 5 above)
Comment (parmarsuraj99) [reply signals/quandl/example_model_quandl.py]: Thanks @LiamHz for the changes :)

File: signals/quandl/example_model_quandl.py [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+        dict(zip(ticker_map["yahoo"], ticker_map["bloomberg_ticker"]))
+    )
+    full_data.sort_index(ascending=True, inplace=True)
+    gc.collect()
+
+    return full_data
+
+
+def main():
+
```
Comment (LiamHz) [inline signals/quandl/example_model_quandl.py:105]:
```suggestion
```

**Review by LiamHz — APPROVED** [review]

**Pull request merged**